api/src/convert/archive_block.rs

```diff
@@ -33,13 +28,25 @@ impl TryFrom<&ArchiveBlock> for BlockData {
     type Error = ConversionError;
 
     fn try_from(src: &ArchiveBlock) -> Result<Self, Self::Error> {
-        if !src.has_v1() {
+        if src.block.is_none() {
```

Comment:
the `has_v1` was checking for both non vs some as well as None, however looking now could probably remove this if condition and only use the match condition below
```rust
    pub fn has_v1(&self) -> bool {
        match self.block {
            ::std::option::Option::Some(ArchiveBlock_oneof_block::v1(..)) => true,
            _ => false,
        }
    }
```

Reply:
removed in https://github.com/mobilecoinfoundation/mobilecoin/pull/4032/commits/c99135ba4c3d5470245130a0b61b58129e52bb3e